atomic_reactor/utils/resource_manager.py

```diff
@@ -67,21 +55,131 @@ def username(self):
     def ssh_keyfile(self):
         return self._ssh_keyfile
 
-    def _validate_resource_lock(self, payload: dict):
-        """ Validate resource lock json file against schema """
-        with open(RESOURCE_LOCK_SCHEMA, 'r') as f:
-            schema_data = f.read()
-        schema = json.loads(schema_data)
+    @property
+    def slots_dir(self):
+        if self._slots_dir is None:
+            home, _, _ = self._run("pwd")
+            self._slots_dir = os.path.join(home, SLOTS_RELATIVE_PATH)
+        return self._slots_dir
+
+    def _get_slot_path(self, slot_id: int) -> str:
+        return os.path.join(self.slots_dir, f"slot_{slot_id}")
+
+    def _is_valid_slot_id(self, slot_id: int) -> bool:
+        valid_slots = range(self.slots)
```

Comment:
```suggestion
        valid_slots = list(range(self.slots))
```

Reply:
Wouldn't leaving at as a `range` object be better? The main purpose of valid_slots is to check if slot_id is in bounds, and range objects are perfect for that (they even have an efficient `__contains__` method)
```python
In [11]: 2**29 in range(2**30)  # don't convert to list unless you want your REPL to get oom-killed
Out[11]: True
```

Reply:
Ah, makes sense. Well, it's not a big deal to convert it to list right away, since slot counts will be small